database_benchmarking_suite/src/main/scala/uk/ac/ebi/eva/benchmarking_suite/cassandra/CassandraReadSampler.scala

```diff
@@ -18,21 +18,22 @@ class CassandraReadSampler() extends AbstractSampler {
       databaseAction = () => {
         val numReadsPerThread = this.getPropertyAsInt("numOpsPerThread")
         val threadNum = this.getThreadContext.getThreadNum
-        randomNumGen = new scala.util.Random(threadNum)
+        //Use thread number and timestamp to vary the random seed across multiple runs for a same thread choice
+        randomNumGen = new scala.util.Random(threadNum  + System.currentTimeMillis())
         (1 to numReadsPerThread).foreach(_ => readData())
       })
 
   }
 
   private def readData(): Unit = {
-    val chromosome = randomNumGen.nextInt(16)
-    val startPos = randomNumGen.nextInt(1e9.toInt/16)
+    val chromosome = randomNumGen.nextInt(32)
+    val startPos = randomNumGen.nextInt(1e8.toInt/32)
     val rows: ResultSet = cassandraTestParams.session.execute(cassandraTestParams.blockReadStmt.bind(
       "eva_hsapiens_grch37",
       chromosome.toString,
       new Integer(startPos),
       new Integer(startPos + blockReadSize)
-    ))
+    ).setReadTimeoutMillis(600000))
```

Comment:
Could this be `10.minutes` like in the rest of the code? I also think that the timeout should be extracted to a global variable.

Reply:
Cassandra benchmarking code uses the Java driver which does not take Scala Duration as an argument. Read timeout may not be reusable other than these 2 databases because I would expect much higher timeouts for Postgres or HBase where scanning for uncached rows may take longer.

Reply:
Fair enough.